pkg/k8shandler/logstore_test.go

```diff
@@ -347,3 +348,77 @@ func TestDifferenceFoundWhenNodeCountExceeds4(t *testing.T) {
 		t.Errorf("Expected that difference would be found due to node count change")
 	}
 }
+
+func TestGenUUIDPreservedWhenNodeCountExceeds4(t *testing.T) {
+	cluster := &logging.ClusterLogging{
+		Spec: logging.ClusterLoggingSpec{
+			LogStore: logging.LogStoreSpec{
+				Type: "elasticsearch",
+				ElasticsearchSpec: logging.ElasticsearchSpec{
+					NodeCount: 3,
+				},
+			},
+		},
+	}
+	elasticsearchCR := newElasticsearchCR(cluster, "test-app-name")
+	dataUUID := esutils.GenerateUUID()
+	elasticsearchCR.Spec.Nodes[0].GenUUID = &dataUUID
+
+	cluster = &logging.ClusterLogging{
+		Spec: logging.ClusterLoggingSpec{
+			LogStore: logging.LogStoreSpec{
+				Type: "elasticsearch",
+				ElasticsearchSpec: logging.ElasticsearchSpec{
+					NodeCount: 4,
+				},
+			},
+		},
+	}
+	elasticsearchCR2 := newElasticsearchCR(cluster, "test-app-name")
+
+	diffCR, different := isElasticsearchCRDifferent(elasticsearchCR, elasticsearchCR2)
+	if !different {
+		t.Errorf("Expected that difference would be found due to node count change")
+	}
+
+	if diffCR.Spec.Nodes[0].GenUUID == nil || *diffCR.Spec.Nodes[0].GenUUID != dataUUID {
+		t.Errorf("Expected that original GenUUID would be preserved as %v but was %v", dataUUID, diffCR.Spec.Nodes[0].GenUUID)
+	}
+}
+
+func TestGenUUIDPreservedWhenNodeCountChanges(t *testing.T) {
```

Comment:
How is this different from the previous test aside from 1->3 and 3->4 ?  What about the code makes this significant?

Reply:
This was the nature of the bz, when we scale between 1 and 3 the UUID was preserved, when we scaled up past 3 to 4 the UUID would disappear. This was due to the fact we were creating a new `node` and not pulling the existing UUID from the CR after doing a comparison to see if the nodes changed.